### DISCUSSION TITLE
What happens if I don't call `initializeMMKV`?

### DISCUSSION THREAD

**Discussion by hoangelec** [Tencent/MMKV#751] [Q&A] [answered]
Hello all, 
I'm using MMKV for my Swift project, and I just realized that in my project I haven't called `initializeMMKV()` at app startup this whole time, however I can still utilize MMKV functions without problems so far. 
So I would like to know the purpose of why we need to `initializeMMKV` as you instructed in the tutorial? 

Thanks in advance

**Comment by lingol** [comment, marked as answer]
You should initialize MMKV in the main thread.
Otherwise, chances are you might get **the first MMKV instance** in a **child thread**, and that will trigger a runtime warning about accessing UIKit properties in child threads.
